src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Operation/NewAzureVMCommand.cs

```diff
@@ -162,10 +186,18 @@ public class NewAzureVMCommand : VirtualMachineBaseCmdlet
             "Win2012Datacenter",
             "Win2008R2SP1")]
         public string ImageName { get; set; } = "Win2016Datacenter";
+        
+        [Parameter(ParameterSetName = DiskFileParameterSet, Mandatory = true)]
+        [ValidateNotNullOrEmpty]
+        public string DiskFile { get; set; }
```

Comment:
You need to update \help\New-AzureRmVM.md file for new parameters.

Reply:
ok

Reply:
Then the parameter set would be mixed for image & disk file cases. I don't think it's actually a good way to do so. We may expand the DiskFile set later on, as well.